src/inttest/java/com/faforever/api/avatar/AvatarControllerTest.java

```diff
@@ -0,0 +1,146 @@
+package com.faforever.api.avatar;
+
+import com.faforever.api.AbstractIntegrationTest;
+import com.faforever.api.data.domain.Avatar;
+import com.faforever.api.security.OAuthScope;
+import com.faforever.api.utils.FileHandlingHelper;
+import org.junit.After;
+import org.junit.Test;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.http.MediaType;
+import org.springframework.mock.web.MockMultipartFile;
+import org.springframework.security.test.context.support.WithUserDetails;
+import org.springframework.test.context.jdbc.Sql;
+import org.springframework.test.context.jdbc.Sql.ExecutionPhase;
+import org.springframework.test.web.servlet.request.MockMultipartHttpServletRequestBuilder;
+import org.springframework.test.web.servlet.request.MockMvcRequestBuilders;
+
+import java.io.IOException;
+import java.nio.file.Files;
+import java.nio.file.Paths;
+import java.util.Optional;
+
+import static org.hamcrest.MatcherAssert.assertThat;
+import static org.hamcrest.core.Is.is;
+import static org.springframework.test.web.servlet.request.MockMvcRequestBuilders.delete;
+import static org.springframework.test.web.servlet.result.MockMvcResultMatchers.content;
+import static org.springframework.test.web.servlet.result.MockMvcResultMatchers.status;
+
+@Sql(executionPhase = ExecutionPhase.BEFORE_TEST_METHOD, scripts = "classpath:sql/prepDefaultUser.sql")
+@Sql(executionPhase = ExecutionPhase.BEFORE_TEST_METHOD, scripts = "classpath:sql/prepAvatarData.sql")
+@Sql(executionPhase = ExecutionPhase.AFTER_TEST_METHOD, scripts = "classpath:sql/cleanAvatarData.sql")
+public class AvatarControllerTest extends AbstractIntegrationTest {
+
+  @Autowired
+  AvatarRepository avatarRepository;
+
+  @Test
+  @WithUserDetails(AUTH_MODERATOR)
+  public void moderatorCanUpload() throws Exception {
+    mockMvc.perform(
+      createAvatarUploadRequest()
+        .with(getOAuthToken(OAuthScope._UPLOAD_AVATAR))
+    ).andExpect(status().isCreated())
+      .andExpect(content().string(""));
+    final Avatar avatar = avatarRepository.findOneByUrl("http://localhost/faf/avatars/avatar3.png").get();
+    assertThat(avatar.getUrl(), is("http://localhost/faf/avatars/avatar3.png"));
+    assertThat(avatar.getTooltip(), is("Best avatar"));
```

Comment:
We should assert that the files exists on the file system.

Reply:
this is tested in unit test of AvatarService